[DOCS] Consolidates ELSER deployment guide with ES deployment tutorial content

szabosteve opened the pull request:
## Overview

Related to https://github.com/elastic/platform-docs-team/issues/237 and https://github.com/elastic/enterprise-search-team/issues/5901.

This PR adds the majority of the [ELSER-related deployment content](https://www.elastic.co/guide/en/enterprise-search/current/elser-text-expansion.html) from the Enterprise Search book to the ELSER conceptual docs, it also adds a new section to the page called `Performance considerations` that copies the relevant information to the page from the [`Limitations and notes`](https://www.elastic.co/guide/en/enterprise-search/current/elser-text-expansion.html#elser-limitations) section of the ES page.

### Preview

[ELSER ES](https://stack-docs_2560.docs-preview.app.elstc.co/guide/en/machine-learning/master/ml-nlp-elser.html#elasticsearch)

chriscressman commented on docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:
```diff
@@ -113,7 +114,7 @@ image::images/ml-nlp-deployment-id-elser-v2.png[alt="Deploying ELSER",align="cen
+[[elasticsearch]]
 === Using the Indices page in {ents}
 
 You can also download and deploy ELSER to an {infer} pipeline directly from the
```
Suggestion:

```asciidoc
Alternatively, you can download and deploy ELSER to an {infer} pipeline using the search indices UI.
```

chriscressman commented on docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:
```diff
@@ -113,7 +114,7 @@ image::images/ml-nlp-deployment-id-elser-v2.png[alt="Deploying ELSER",align="cen
 [discrete]
-[[enterprise-search]]
+[[elasticsearch]]
 === Using the Indices page in {ents}
```
```suggestion
=== Using the search indices UI
```

chriscressman commented on docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:
```diff
@@ -125,10 +126,54 @@ to use ELSER.
+To add ELSER to an ingest pipeline, you need to copy the default ingest 
+pipeline and then customize it according to your needs.
+
+1. Click **Copy and customize** under the **Unlock your custom pipelines** block 
```
Suggestion:

```asciidoc
1. Click **Copy and customize** under the **Unlock your custom pipelines** block 
at the top of the page. This enables the **Add inference pipeline** button.
```

chriscressman reviewed (commented):
@szabosteve These changes look great.

I suggested a few changes inline. There are also two instances where the Kibana breadcrumb needs to be updated. The breadcrumb used to be "Enterprise Search > Indices", but now it is "Search > Indices". I forgot to call this out in the requirements.

I didn't spot any other instances of "Enterprise Search" in the doc, but in general we want to get rid of any mentions of Enterprise Search and instead present this as a feature specific to Elasticsearch and Kibana.

szabosteve commented:
@chriscressman Thank you for your thoughtful review; I updated the breadcrumbs and addressed your comments via [f02d5ee](https://github.com/elastic/stack-docs/pull/2560/commits/f02d5eee0106d863fe299cdc344f23a370c23f99).

leemthompo reviewed (approved):
Nice, thanks for driving this one!
